client/my-sites/checkout/upsell-nudge/professional-email-upsell/index.tsx

```diff
@@ -137,16 +168,35 @@ const ProfessionalEmailUpsell = ( {
 		};
 
 		const cartItem =
-			productSlug === TITAN_MAIL_MONTHLY_SLUG
+			selectedIntervalLength === IntervalLength.MONTHLY
 				? titanMailMonthly( cartItemArgs )
 				: titanMailYearly( cartItemArgs );
 
 		setCartItem( cartItem, () => handleClickAccept( 'accept' ) );
 	};
 
+	const threeMonthsFreeBadge = (
+		<span>
+			<Badge type="success">{ translate( '3 months free' ) }</Badge>
+		</span>
+	);
+
+	const pricingComponent = (
+		<div className="professional-email-upsell__pricing">
+			{ threeMonthsFreeBadge }
```

Comment:
Given that this is the only place we use `threeMonthsFreeBadge`, it may be worth inlining the elements from above so we have one block of JSX instead of two. (Definitely not blocking!)
```suggestion
			<span>
				<Badge type="success">{ translate( '3 months free' ) }</Badge>
			</span>
```